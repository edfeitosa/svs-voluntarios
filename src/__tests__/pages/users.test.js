import React from 'react';
import { MemoryRouter as Router, withRouter } from 'react-router-dom';
import User from './../../pages/user';

describe('user component', () => {

    let wrapper;
    beforeEach(() => wrapper = mount(
        <Router>
            <User />
        </Router>
    ));

    test('renders without crashing', () => {
        const wrapper = renderer
            .create(
                <Router>
                    <User />
                </Router>
            )
            .toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    test('initial state', () => {
        const wrapper = mount(<User />);
        expect(wrapper.state().showModal).toEqual(false);
        wrapper.unmount();
    });

    test('simulate click for change state', () => {
        wrapper.find('.new-buttom').simulate('click', (
            wrapper.setState({ showModal: true })
        ));
        expect(wrapper.state().showModal).toEqual(true);
        wrapper.unmount();
    });

});