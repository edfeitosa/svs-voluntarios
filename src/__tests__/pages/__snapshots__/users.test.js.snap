// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user component renders without crashing 1`] = `
<div
  className="padding-content"
>
  <div>
    <h3>
      USU√ÅRIOS
    </h3>
    <div
      className="new-buttom"
      onClick={[Function]}
    >
      <img
        alt="Novo"
        src="test-file-stub"
        title="Novo"
      />
    </div>
  </div>
</div>
`;
